CONTRIBUTING.md

```diff
@@ -46,18 +44,26 @@ Setup guides, FAQ, the more information we have on the [wiki](https://wiki.kavit
 - Only make pull requests to develop, never main, if you make a PR to main we'll comment on it and close it
 - You're probably going to get some comments or questions from us, they will be to ensure consistency and maintainability
 - We'll try to respond to pull requests as soon as possible, if its been a day or two, please reach out to us, we may have missed it
-- Each PR should come from its own [feature branch](http://martinfowler.com/bliki/FeatureBranch.html) not develop in your fork, it should have a meaningful branch name (what is being added/fixed)
+- Each PR should come from its own [feature branch](http://martinfowler.com/bliki/FeatureBranch.html), not the `develop` branch in your fork, and should have a meaningful branch name (what is being added/fixed)
     - new-feature (Good)
     - fix-bug (Good)
     - patch (Bad)
     - develop (Bad)
     - feature/parser-enhancements (Great)
     - bugfix/book-issues (Great)
 
+### Testing ###
+We use [Cypress](https://www.cypress.io/) for end-to-end testing before each release
+- Make sure the API is running in Visual Studio/Rider before continuing
+- To open the Cypress GUI, run `ng e2e`, then go to `Start E2E Testing > Specs`. The specs can be found in `Kavita\UI\Web\cypress\e2e`
+- To run all specs in order from the command line, run `ng run kavita-webui:cypress-run --browser firefox`
+  - If you don't have Firefox installed, substitute `--browser firefox` with `--browser edge` or `--browser chrome`
+- To run a specific spec from the command line, run `ng run kavita-webui:cypress-run --browser firefox --spec "cypress\e2e\SPEC_FILE"`
+
 ### Swagger API ###
 If you just want to play with Swagger, you can just
```

Comment:
Maybe add that it can be enabled from the Kavita instance without the need to build the whole Kavita?